docs: clarify some details in api style guide (#9843)
zitadel/zitadel: API_DESIGN.md

Additionally, there are some conventions that are specific to the ZITADEL API.
These conventions are described in the following sections.

### File Structure

> In addition to the files structure proposed by Protobuf and connectRPC, the ZITADEL API uses the following order for `Everything else` section of files:

Reviewer: ```suggestion
In addition to the files structure proposed by [Protobuf](https://protobuf.dev/programming-guides/style/#file-structure) and [connectRPC](https://buf.build/docs/best-practices/style-guide/#recommendations), the ZITADEL API uses the following order for `Everything else` section of files:
```

- `Service(s)`

> - `Messages` / `Enums`

Reviewer: would it make sense to add some info about ordering of requests ? e.g. reading > creation > update > remove

and mention that messages should have the order of the requests

Please check out the structure Buf style guide for more information about the folder and package structure: https://buf.build/docs/best-practices/style-guide/

#### Removal and Deprecation of Message Fields

When ever a message field is not supported anymore, deprecation of the field is preferred over removal.
This prevents breaking changes and allows the client to handle the deprecation gracefully.

> The field should be marked as deprecated and the reason for the deprecation should be documented in the proto file.

Reviewer: ```suggestion
The field MUST be marked as deprecated and the reason for the deprecation MUST be documented in the proto file as well as the major version the field will be removed. 
```

} 
```

> Return messages should only contain the fields that are required to be returned. Endpoints manipulating or deleting a resource SHOULD whenever possible only return the change or deletion date.

Reviewer: ```suggestion
Return messages SHOULD only contain the fields that are required to be returned. Endpoints manipulating or deleting a resource SHOULD whenever possible only return the change or deletion date.
```
